flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/api/TableConfig.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.api;
+
+import org.apache.flink.api.common.time.Time;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.GlobalConfiguration;
+import org.apache.flink.util.Preconditions;
+
+import java.math.MathContext;
+import java.util.TimeZone;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+
+/**
+ * A config to define the runtime behavior of the Table API.
+ */
+public class TableConfig {
```

Comment:
I think TableConfig and PlannerConfig should stay in planner module

Reply:
I moved them to runtime module because some runtime operator implementation depends on it.  But now, I decoupled the implementation from `TableConfig`. So I think it makes sense to keep it in planner module for now. 

However, in the end, the TableConfig should be moved to api module. And actually, it is been moved to api module in the PR #8050 .